sql/analyzer/resolve_columns.go

```diff
@@ -104,20 +104,25 @@ type column interface {
 // nestingLevelSymbols tracks available table and column name symbols at a nesting level for a query. Each nested
 // subquery represents an additional nesting level.
 type nestingLevelSymbols struct {
```

Comment:
These two types probably belong in your new file as well

Reply:
none of these are used for projection pruning